lib/ecl/ecl_sum_file_data.cpp

```diff
@@ -0,0 +1,736 @@
+}
+
+
+// ***************************** End writing *************************************
+
+// **************************** Start Reading ************************************
+
+bool ecl_sum_file_data::check_file( ecl_file_type * ecl_file ) {
+  if (ecl_file_has_kw( ecl_file , PARAMS_KW ) &&
+      (ecl_file_get_num_named_kw( ecl_file , PARAMS_KW ) == ecl_file_get_num_named_kw( ecl_file , MINISTEP_KW)))
+    return true;
+  else
+    return false;
+}
+
+/**
+   Malformed/incomplete files:
+   ----------------------------
+   Observe that ECLIPSE works in the following way:
+
+     1. At the start of a report step a summary data section
+        containing only the 'SEQHDR' keyword is written - this is
+        currently an 'invalid' summary section.
+
+     2. ECLIPSE simulates as best it can.
+
+     3. When the time step is complete data is written to the summary
+        file.
+
+   Now - if ECLIPSE goes down in flames during step 2 a malformed
+   summary file will be left around, to handle this situation
+   reasonably gracefully we check that the ecl_file instance has at
+   least one "PARAMS" keyword.
+
+   One ecl_file corresponds to one report_step (limited by SEQHDR); in
+   the case of non unfied summary files these objects correspond to
+   one BASE.Annnn or BASE.Snnnn file, in the case of unified files the
+   calling routine will read the unified summary file partly.
+*/
+
+void ecl_sum_file_data::add_ecl_file(int report_step, const ecl_file_view_type * summary_view, const ecl_smspec_type * smspec) {
+
+  int num_ministep  = ecl_file_view_get_num_named_kw( summary_view , PARAMS_KW);
+  if (num_ministep > 0) {
+    int ikw;
+
+    for (ikw = 0; ikw < num_ministep; ikw++) {
+      ecl_kw_type * ministep_kw = ecl_file_view_iget_named_kw( summary_view , MINISTEP_KW , ikw);
+      ecl_kw_type * params_kw   = ecl_file_view_iget_named_kw( summary_view , PARAMS_KW   , ikw);
+
+      {
+        int ministep_nr = ecl_kw_iget_int( ministep_kw , 0 );
+        ecl_sum_tstep_type * tstep = ecl_sum_tstep_alloc_from_file( report_step ,
+                                                                    ministep_nr ,
+                                                                    params_kw ,
+                                                                    ecl_file_view_get_src_file( summary_view ),
+                                                                    smspec );
+
+        if (tstep)
+            append_tstep( tstep );
+      }
+    }
+  }
+}
+
+
+bool ecl_sum_file_data::fread(const stringlist_type * filelist) {
+  if (stringlist_get_size( filelist ) == 0)
+    return false;
+
+  {
+    ecl_file_enum file_type = ecl_util_get_file_type( stringlist_iget( filelist , 0 ) , NULL , NULL);
+    if ((stringlist_get_size( filelist ) > 1) && (file_type != ECL_SUMMARY_FILE))
+      util_abort("%s: internal error - when calling with more than one file - you can not supply a unified file - come on?! \n",__func__);
+
+    {
```

Comment:
This is one of these guards that immediately follows a brace-less `if` sentence which is really dangerous in my opinion.  It seems the scope below belongs to the `if`.

Furthermore, if you put the preceding `int filenr` into the `for` loop, there is nothing that leaks out anyway, so the guard can safely be deleted.

Reply:
Again - very old code.